R/get_ref.R

```diff
@@ -414,11 +418,62 @@ get_release_date.remote_ref <- function(remote_ref) {
   NA
 }
 
+#' Get the pseudo-published date of Bioc packages
+#'
+#' The date from the last modification to the sources is assumed to be the
+#' published date as it is not available in `pkgcache` results.
+#'
+#' The official metadata of the Bioconductor release does not work for the
+#' purpose of this package (https://bioconductor.org/config.yaml). As there
+#' might be a edge case where a Bioconductor package was changed and depends
+#' on a newer version of a package. (Actual edge case with SummarizedExperiment
+#' and matrixStats)
+#'
+#' @keywords internal
+get_bioc_package_release_date <- function(package) {
+  pkg_list <- pkgcache::meta_cache_list(packages = package)
+  tryCatch({
+    pkg_list <- pkg_list[order(package_version(pkg_list$version)),][1,]
+    source_url <- pkg_list$sources[[1]]
+    last_modified <- httr::GET(source_url)$headers$`last-modified`
+    as.POSIXct(strptime(last_modified, format = "%a, %d %b %Y %H:%M:%S"))
+  },
+  error = function(err) {
+    cli::cli_alert_info(paste0(
+      "Couldn't find the release date for ",
+      cli::col_blue("bioc::{{{package}}}"),
+      ". This may cause a failure to install packages as",
+      " min_cohort / min_isolated strategies depend on the release date."
+    ))
+    NA
+  })
+}
```

Comment:
Problem with Bioconductor versions as they don't have any metadata information about the publication date.

```R
r> pkgcache::meta_cache_list("SummarizedExperiment")$published
[1] NA
```

**This is very open to change!!**

Workaround checks the last modified date of the package source.

However, I couldn't find an API that would retrieve the information and using Bioconductor release date doesn't work as specific edge case has 2 packages with new releases _(where 1 depends on the other)_.